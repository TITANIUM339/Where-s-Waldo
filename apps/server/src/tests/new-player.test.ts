import express from "express";
import request from "supertest";
import { describe, expect, test } from "vitest";
import errorHandler from "../middleware/error-handler";
import newPlayerRouter from "../routes/new-player/new-player";

const app = express();

app.use(newPlayerRouter);

app.use(errorHandler);

describe("responds with 400 for bad input", () => {
    test("GET /new-player", async () => {
        const response = await request(app).get("/new-player");

        expect(response.status).toBe(400);
        expect(response.body).toMatchSnapshot();
    });

    test("GET /new-player?name=This string is too long", async () => {
        const response = await request(app).get(
            "/new-player?name=This string is too long",
        );

        expect(response.status).toBe(400);
        expect(response.body).toMatchSnapshot();
    });
});

describe("responds with 200 for correct input", () => {
    test("GET /new-player?name=Player", async () => {
        const response = await request(app).get("/new-player?name=Player");

        expect(response.status).toBe(200);
        expect(response.body).toMatchObject({
            token: /^[\w-]+\.[\w-]+\.[\w-]+$/,
        });
    });
});
