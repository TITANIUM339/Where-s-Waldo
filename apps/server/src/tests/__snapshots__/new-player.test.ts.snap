// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`responds with 400 for bad input > GET /new-player 1`] = `
{
  "data": [
    {
      "location": "query",
      "msg": "required",
      "path": "name",
      "type": "field",
      "value": "",
    },
  ],
  "message": "Bad Request",
  "status": 400,
}
`;

exports[`responds with 400 for bad input > GET /new-player?name=This string is too long 1`] = `
{
  "data": [
    {
      "location": "query",
      "msg": "maximum length is 16",
      "path": "name",
      "type": "field",
      "value": "This string is too long",
    },
  ],
  "message": "Bad Request",
  "status": 400,
}
`;
